core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1981,29 +1986,39 @@ class KafkaApis(val requestChannel: RequestChannel,
     val results = new DeletableTopicResultCollection(deleteTopicRequest.data.topicNames.size)
     val toDelete = mutable.Set[String]()
     if (!controller.isActive) {
-      deleteTopicRequest.data.topicNames.forEach { topic =>
+      deleteTopicRequest.topics().forEach { topic =>
         results.add(new DeletableTopicResult()
-          .setName(topic)
+          .setName(topic.name())
+          .setTopicId(topic.topicId())
           .setErrorCode(Errors.NOT_CONTROLLER.code))
       }
       sendResponseCallback(results)
     } else if (!config.deleteTopicEnable) {
       val error = if (request.context.apiVersion < 3) Errors.INVALID_REQUEST else Errors.TOPIC_DELETION_DISABLED
-      deleteTopicRequest.data.topicNames.forEach { topic =>
+      deleteTopicRequest.topics().forEach { topic =>
         results.add(new DeletableTopicResult()
-          .setName(topic)
+          .setName(topic.name())
+          .setTopicId(topic.topicId())
           .setErrorCode(error.code))
       }
       sendResponseCallback(results)
     } else {
-      deleteTopicRequest.data.topicNames.forEach { topic =>
+      deleteTopicRequest.topics().forEach { topic =>
+        val name = if (topic.name() != null) topic.name() 
```

Comment:
According to the kip, we will use topicId firstly and topic name will be used only if topicId is ZERO, but you just use the topic name even topicId is not ZERO here?

Reply:
I'm setting both topic ID and name here for the result since we will return with both, but if you look at the code below, it handles topic ID errors first. From then on, I'm not sure if it makes a difference. I suppose I could check if the topic ID is zero rather than if the name is null, but right now the code paths either set topic ID and name to null or set topic name and topic ID to zero UUID. So for now, those do equivalent things.

Reply:
I'll switch to checking for zero UUID for now.